ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoLedgerService.kt

```diff
@@ -38,5 +43,54 @@ interface UtxoLedgerService {
     @Suspendable
     fun <T : ContractState> resolve(stateRef: StateRef): StateAndRef<T>
 
+    /**
+     * Finds a transaction by id in the vault and returns it
+     *
+     * @param id The id of the transaction to find
+     *
+     * @return The signed transaction, if it has been recorded previously. Null if not found.
+     */
+    @Suspendable
+    fun findSignedTransaction(id: SecureHash): UtxoSignedTransaction?
+
+    /**
+     * Finds a transaction by id in the vault, resolves it to a ledger transaction and returns it
+     *
+     * @param id The id of the transaction to find
+     *
+     * @return The ledger transaction, if it has been recorded previously. Null if not found.
+     */
+    @Suspendable
+    fun findLedgerTransaction(id: SecureHash): UtxoLedgerTransaction?
+
+    /**
+     * Collects signatures, records and broadcasts to involved peers a [UtxoSignedTransaction].
+     *
+     * @param signedTransaction The [UtxoSignedTransaction] to finalise and recorded locally and with peer [sessions].
+     * @param sessions The [FlowSession]s of the peers involved in the transaction.
+     *
+     * @return The fully signed [UtxoSignedTransaction] that was recorded.
+     */
+    @Suspendable
+    fun finality(
```

Comment:
See comment above re. `finality` vs. `finalize`.

Reply:
I have renamed finality() to finalize(). (I did not touch receiveFinality()... Let me know if you would prefer to change that too.)